kie-server-parent/kie-server-tests/kie-server-integ-tests-optaplanner/src/test/java/org/kie/server/integrationtests/optaplanner/OptaplannerIntegrationTest.java

```diff
@@ -615,6 +644,77 @@ public void testTerminateEarly() throws Exception {
+                    .connectTimeout(10, TimeUnit.SECONDS)
+                    .readTimeout(10, TimeUnit.SECONDS)
+                    .build();
+
+        WebTarget webTarget = httpClient.target(uriString);
+        webTarget.register(new BasicAuthentication(TestConfig.getUsername(), TestConfig.getPassword()));
+        return webTarget;
+    }
+
+    private void checkPrometheusMetrics() {
+        Response response = null;
+        try {
+            String uriString = TestConfig.getKieServerHttpUrl().replaceAll("/server", "") + "/metrics";
+            WebTarget clientRequest = newRequest(uriString);
+            response = clientRequest.request(MediaType.TEXT_PLAIN_TYPE).get();
+
+            Assert.assertEquals(Response.Status.OK.getStatusCode(), response.getStatus());
+
+            String res = response.readEntity(String.class);
+
+            //These are Prometheus metrics attributes we defined in Optaplanner
+            Assert.assertThat(res, containsString("solvers_running"));
+            Assert.assertThat(res, containsString("solver_duration_seconds_count{solver_id=\"cloudsolver-prometheus\""));
+            Assert.assertThat(res, containsString("solver_duration_seconds_sum{solver_id=\"cloudsolver-prometheus\""));
+
+            Assert.assertThat(res, containsString("solver_score_calculation_speed_count{solver_id=\"cloudsolver-prometheus\""));
+            Assert.assertThat(res, containsString("solver_score_calculation_speed_sum{solver_id=\"cloudsolver-prometheus\""));
+
+            //verify metric values
+            try (BufferedReader br = new BufferedReader(new StringReader(res))) {
+                String line = br.readLine();
+                while (line != null) {
+                    checkPrometheusMetricsLine(line);
+                    line = br.readLine();
+                }
+            }
+
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        } finally {
+            if (response != null) {
+                response.close();
+            }
+        }
+    }
+
+    private void checkPrometheusMetricsLine(String line) {
+        if (line.startsWith("solver_duration_seconds_count")
+                || line.startsWith("solver_duration_seconds_sum")
+                || line.startsWith("solver_score_calculation_speed_count")
+                || line.startsWith("solver_score_calculation_speed_sum")
+        ) {
+            // all metric values should be a positive double
+            checkMetric(line, 1);
+        } else if (line.startsWith("solvers_running")) {
+            //when test is done we should have zero solvers running
+            checkMetric(line, 0);
+        }
+
+    }
+
+    private void checkMetric(String line, double value) {
+        StringTokenizer st = new StringTokenizer(line, " ");
+        Assert.assertEquals(2, st.countTokens());
+        st.nextToken();
+        Assert.assertTrue(Double.parseDouble(st.nextToken()) >= value);
```

Comment:
Generally it's good to avoid assertTrue as much as possible, because it doesn't tell much. 

Assert.equals or Assert.greaterThan serve much better, as they inform you about what value differed from the expected value. With assertTrue, you have only true/false.